stbt_rig.py

```diff
@@ -1213,11 +1214,37 @@ def _post(self, suffix="", **kwargs):
             "/api/v2/nodes/%s/%s" % (self.node_id, suffix), **kwargs)
 
 
+_USER_AGENT = None
+
+
+def user_agent(mode):
+    global _USER_AGENT  # pylint: disable=global-statement
+    if not _USER_AGENT:
+        with open(__file__.rstrip('c'), "rb") as f:
+            my_src = f.read()
+        my_src = my_src.replace(b"\r\n", b"\n")
+        s = hashlib.sha1(b"blob %i\0" % len(my_src))
+        s.update(my_src)
+        git_blob_sha = s.hexdigest()[:7]
+
+        if "VSCODE_PID" in os.environ:
+            ide = "; ide:vscode"
+        elif "PYCHARM_HOSTED" in os.environ:
+            ide = "; ide=pycharm"
```

Comment:
Is there any kind of standard for user-agent strings regarding `:` vs. `=` vs. spaces?

Reply:
Firefox uses colons at least: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent/Firefox